src/sync.ts

```diff
@@ -992,13 +993,13 @@ export class SyncApi {
             pollTimeout = 0;
         }
 
-        let filterId = syncOptions.filterId;
-        if (this.client.isGuest() && !filterId) {
-            filterId = this.getGuestFilter();
+        let filter = syncOptions.filter;
```

Comment:
```suggestion
        const filter = syncOptions.filter || (this.client.isGuest() && this.getGuestFilter());
```

Reply:
We don't use this style anywhere else in the codebase, and IMO it is harder to read quickly

Reply:
Readability is subjective but noted.

Consider the upside of reducing the amount of mutable state in scope.